File: src/main/java/org/proofit/premiumcalc/model/Policy.java
package org.proofit.premiumcalc.model;

import java.util.List;
import java.util.Objects;

public final class Policy {
    private final String policyNumber;
    private final PolicyStatus status;
    private final List<PolicyObject> objects;

    public Policy(String policyNumber, PolicyStatus status, List<PolicyObject> objects) {
        this.policyNumber = policyNumber;
        this.status = status;
        this.objects = objects;
    }

    public List<PolicyObject> getObjects() {
        return objects;
    }
}

File: src/main/kotlin/org/proofit/premiumcalc/model/Policy.kt
package org.proofit.premiumcalc.model

import java.math.BigDecimal

data class Policy(
        val policyNumber: String,
        val status: PolicyStatus,
        val objects: List<PolicyObject>
)

data class PolicyObject(
        val objectName: String,
        val subObjects: List<PolicySubObject>
)

data class PolicySubObject(
        val subObjectName: String,
        val sumInsured: BigDecimal,
        val riskType: RiskType
)


enum class PolicyStatus {
    REGISTERED, APPROVED
}

enum class RiskType {
    FIRE, THEFT
}
